forte/processors/data_augment/data_select_index_pipeline.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=useless-super-delegation
+
+"""
+This class is used for data selection of the data augmentation tasks.
+For each specific dataset, we first create a reader that reads
+in the document text as Datapack.
+Then we pipeline the reader and the ElasticSearchIndexProcessor
+to create a indexer,
+which is defined in forte/data/indexers/EleasticSearchIndexer.
+This class is then used by the DataSelector class to search for documents.
+"""
+
+from typing import Dict, Any
+
+from forte.common.configuration import Config
+from forte.data.readers.base_reader import PackReader
+from forte.data.data_pack import DataPack
+from forte.data.readers import MSMarcoPassageReader
+from forte.indexers.elastic_indexer import ElasticSearchIndexer
+from forte.pipeline import Pipeline
+from forte.processors.data_augment import DataSelectorIndexProcessor
+
+
+__all__ = [
+    "CreateIndexerBasePipeline",
+    "CreateIndexerMSMacroPipeline",
+]
+
+
+class CreateIndexerBasePipeline:
+
+    def __init__(self):
+        self.reader = self._set_reader()
+        self.datapath = self._set_data_path()
+        self.config = Config(self.default_config(), default_hparams=None)
+        self.create_pipeline()
+        self.create_index()
+
+    def _set_reader(self) -> PackReader:
+        """
+        Return: A dataset Reader.
+        """
+        raise NotImplementedError
+
+    def _set_data_path(self) -> str:
```

Comment:
Why don't we just put this as the `default_config`? And when we use this we simply change the config?

Reply:
Actually, I don't find data source is the best way to use this. The source of data (input parameters) should depend on the implementation of the reader. So we cannot assume that a path is enough. Maybe we can expose the parameters in the `create_index` function?